### PR TITLE
Switching task list type int32 to TaskListType_Decision/TaskListType_Activity enum

### PR DESCRIPTION
**Pull request by markmark206**
**What changed?**
Replaced the `int32` value signifying taskListType with an [already existing] enum, `TaskListType`, for cleaner code and stronger type safety. 

**Why?**
We wanted to clean up the code a bit, and make things more maintainable by the virtue of stronger type safety, by using a specific type instead of a generic `int32` to represent task list type, and to remove the duplication of definitions (`TaskListTypeActivity`/`TaskListTypeDecision` vs `TaskListType_Activity` / `TaskListType_Decision`).

**How did you test it?**
Built and ran unit tests, put up a pull request for buildkite to run tests.  

**Potential risks**
This is purely internal code change, and there should be no changes in the behavior. If there is a bug, and things don't work as expected, they will hopefully be caught by our tests, and we will back out the change, or fix the problem.

### CODE REVIEW

File: proto/persistenceblobs/server_message.proto [hunk 1]
```diff
@@ -192,7 +193,7 @@ message AllocatedTaskInfo {
 message TaskListInfo {
     bytes namespaceId = 1;
     string name = 2;
-    int32 taskType = 3;
+    tasklist.TaskListType taskType = 3;
     // {Normal, Sticky}
     int32 kind = 5;
```
Comment (alexshtin) [inline proto/persistenceblobs/server_message.proto:198]: `kind` is the next good candidate.

File: service/matching/config.go [hunk 2]
```diff
@@ -30,6 +30,7 @@ import (
 	"github.com/temporalio/temporal/common"
 	"github.com/temporalio/temporal/common/cache"
 	"github.com/temporalio/temporal/common/service/dynamicconfig"
+	// tasklistpb "go.temporal.io/temporal-proto/tasklist"
```
Comment (alexshtin) [inline service/matching/config.go:33]: Remove this.

File: service/matching/db.go [hunk 3]
```diff
@@ -33,6 +33,7 @@ import (
 	"github.com/temporalio/temporal/common/log/tag"
 	"github.com/temporalio/temporal/common/persistence"
 	"github.com/temporalio/temporal/common/primitives"
+	tasklistpb "go.temporal.io/temporal-proto/tasklist"
```
Comment (alexshtin) [inline service/matching/db.go:36]: I think you need to run `make goimports` to fix ordering here.

File: common/persistence/dataInterfaces.go [hunk 4]
```diff
@@ -124,12 +125,6 @@ const (
 	WorkflowStateCorrupted
 )
 
-// Types of task lists
```
Comment (alexshtin) [inline common/persistence/dataInterfaces.go:127]: :+1:

File: service/matching/config.go (hunk 2 above)
Comment (markmark206) [reply service/matching/config.go]: duh. thanks, removed.

File: proto/persistenceblobs/server_message.proto (hunk 1 above)
Comment (markmark206) [reply proto/persistenceblobs/server_message.proto:198]: great, thank you. queueing this up for the `int`->`enum` work after this pr.

File: service/matching/db.go (hunk 3 above)
Comment (markmark206) [reply service/matching/db.go:36]: hmm... my editor does this automatically, and then i ran `make goimports` by hand, just to make sure, and the order didn't change. are you expecting something different here?

Comment (alexshtin) [reply service/matching/db.go:36]: There should be 3 blocks:
1. system imports
2. third party libs
3. internal packages

`go.temporal.io/temporal-proto/` considered to be 3rd party in this case.

image

Comment (alexshtin) [reply service/matching/db.go:36]: It seems grouping is GoLand feature not goimports.

Comment (alexshtin) [reply service/matching/db.go:36]: You can ignore this then.

**Pull request merged**